File: code/java/scoping-functions/let.java
fina var original = "abc"
// Map the value and send to the next chain

Optional.of(original)
        .map(
            str -> {
                System.out.println("The original String is " + str); // "abc"
                val result = new StringBuilder(input).reverse().toString();
                return result;
            }
        ).map(
            reversed -> {
                System.out.println("The reverse String is " + reversed); // "cba"
                val result = reversed.length();
                return result;
            }
        ).map(
            length -> {
                System.out.println("The length of the String is " + length); // 3
                return length;
            }
        )

File: code/kotlin/scoping-functions/let.kt
val original = "abc"
// Map the value and send to the next chain
original.let {
    println("The original String is $it") // "abc"
    it.reversed() // map it as parameter to send to next let
}.let {
    println("The reverse String is $it") // "cba"
    it.length  // can be evolved to other type
}.let {
    println("The length of the String is $it") // 3
}